code/modules/mob/new_player/preferences_setup.dm

```diff
@@ -207,748 +207,71 @@
-				switch(backbag)
-					if(2)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "backpack"), ICON_OVERLAY)
-					if(3)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel-norm"), ICON_OVERLAY)
-					if(4)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-			if(BLUESHIELD)
-				clothes_s = new /icon(uniform_dmi, "officer_s")
-				clothes_s.Blend(new /icon('icons/mob/feet.dmi', "jackboots"), ICON_UNDERLAY)
-				clothes_s.Blend(new /icon('icons/mob/hands.dmi', "swat_gl"), ICON_UNDERLAY)
-				clothes_s.Blend(new /icon('icons/mob/suit.dmi', "blueshield"), ICON_OVERLAY)
-				switch(backbag)
-					if(2)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "securitypack"), ICON_OVERLAY)
-					if(3)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel-norm"), ICON_OVERLAY)
-					if(4)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-			if(JUDGE)
-				clothes_s = new /icon(uniform_dmi, "really_black_suit_s")
-				clothes_s.Blend(new /icon('icons/mob/feet.dmi', "laceups"), ICON_UNDERLAY)
-				clothes_s.Blend(new /icon('icons/mob/head.dmi', "mercy_hood"), ICON_UNDERLAY)
-				clothes_s.Blend(new /icon('icons/mob/suit.dmi', "judge"), ICON_UNDERLAY)
-				switch(backbag)
-					if(2)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "backpack"), ICON_OVERLAY)
-					if(3)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel-norm"), ICON_OVERLAY)
-					if(4)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-
-	if(disabilities & NEARSIGHTED)
-		preview_icon.Blend(new /icon('icons/mob/eyes.dmi', "glasses"), ICON_OVERLAY)
-
-	// Observers get tourist outfit.
-	if(for_observer)
-		clothes_s = new /icon(uniform_dmi, "tourist_s")
-		clothes_s.Blend(new /icon('icons/mob/feet.dmi', "black"), ICON_UNDERLAY)
-		if(backbag == 2)
-			clothes_s.Blend(new /icon('icons/mob/back.dmi', "backpack"), ICON_OVERLAY)
-		else if(backbag == 3 || backbag == 4)
-			clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-
-	if(underwear_s)
-		preview_icon.Blend(underwear_s, ICON_OVERLAY)
-	if(undershirt_s)
-		preview_icon.Blend(undershirt_s, ICON_OVERLAY)
-	if(socks_s)
-		preview_icon.Blend(socks_s, ICON_OVERLAY)
-	if(clothes_s)
-		preview_icon.Blend(clothes_s, ICON_OVERLAY)
-	preview_icon.Blend(face_s, ICON_OVERLAY)
-	preview_icon_front = new(preview_icon, dir = SOUTH)
-	preview_icon_side = new(preview_icon, dir = WEST)
-
-	qdel(face_s)
-	qdel(underwear_s)
-	qdel(undershirt_s)
-	qdel(socks_s)
-	qdel(clothes_s)
+				preview_icon = icon('icons/mob/robots.dmi', "robot", SOUTH)
+				preview_icon.Scale(64, 64)
+				return
+
+	// Set up the dummy for its photoshoot
+	var/mob/living/carbon/human/dummy/mannequin = new()
+	mannequin.status_flags |= GODMODE // Why not?
+	copy_to(mannequin)
+
+	// Determine what job is marked as 'High' priority, and dress them up as such.
+	var/datum/job/previewJob

```

Comment:
Perhaps this entire section should be broken out into its own proc - the code here is tailored to the task of finding which job is at "high" pref on a character, and it being implemented inside here instead of outside adds a lot of elements to the system that could be difficult to predict